libs/shared/console-shared/src/lib/job-configure-settings/job-configure-settings.tsx

```diff
@@ -0,0 +1,188 @@
+import cronstrue from 'cronstrue'
+import { useEffect, useState } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
+import { JobConfigureData } from '@qovery/shared/interfaces'
+import { InputText, Link } from '@qovery/shared/ui'
+import EnableBox from '../enable-box/enable-box'
+import EntrypointCmdInputs from '../entrypoint-cmd-inputs/entrypoint-cmd-inputs'
+
+export interface JobConfigureSettingsProps {
+  jobType: 'cron' | 'lifecycle'
+}
+
+export function JobConfigureSettings(props: JobConfigureSettingsProps) {
+  const { control, setValue, getValues, watch } = useFormContext<JobConfigureData>()
+  const watchSchedule = watch('schedule')
+  const [cronDescription, setCronDescription] = useState('')
+
+  useEffect(() => {
+    if (watchSchedule) {
+      // check if watchSchedule is a valid cron expression
+      const isValidCron = cronstrue.toString(watchSchedule, { throwExceptionOnParseError: false })
+      if (isValidCron.indexOf('An error') === -1) {
+        setCronDescription(isValidCron)
+      }
+    }
+  }, [watchSchedule])
+
+  return (
+    <div>
+      {props.jobType === 'cron' ? (
+        <>
+          <h3 className="text-sm font-semibold mb-3">CRON</h3>
+          <Controller
+            name="schedule"
+            control={control}
+            rules={{
+              required: 'Value required',
+            }}
+            render={({ field, fieldState: { error } }) => (
+              <InputText
+                className="mb-2"
+                name={field.name}
+                onChange={field.onChange}
+                value={field.value}
+                label="Schedule - Cron expression"
+                error={error?.message}
+              />
+            )}
+          />
+          <div className="mb-3 flex justify-between">
+            <p className="text-text-500 text-xs">{cronDescription}</p>
+            <Link
+              external={true}
+              link="https://crontab.guru/"
+              className="text-text-400 !text-xs"
+              linkLabel="CRON expression builder"
+            />
+          </div>
+          <EntrypointCmdInputs />
+        </>
+      ) : (
+        <div className="mb-10">
+          <h3 className="text-sm font-semibold mb-1">Event</h3>
+          <p className="text-text-500 text-sm mb-3">
+            Select one or more environment event where the job should be executed
+          </p>
+
+          <EnableBox
+            className="mb-3"
+            checked={getValues().on_start?.enabled || false}
```

Comment:
On the top you can decomposed the getValues like this `const { on_start, on_stop } = props` it might be more clean to read